test/e2e/operator_expectations.go

```diff
@@ -121,3 +126,48 @@ func ExpectAllMachinesLinkedToANode() error {
 	}
 	return nil
 }
+
+func ExpectReconcileControllersDeployment() error {
```

Comment:
Consider turning the function into a method and accessing the `F` through method receiver. E.g.
```go
func (TestConfig tc) ExpectReconcileControllersDeployment() error {
...
tc.F.Client...
...
}
```

Reply:
Thanks, done. To give some context @frobware and I discussed about having this as go test framework functions which seemed a bit odd. How ever I'd like to keep this as simple as possible rather than building a custom framework. We might need to reconsider an existing framework if we feel the needs

Reply:
Agree. The framework we have currently in the CI was needed because of non-trivial complex logic that is better wrapped in auxiliary methods. So it can be reused among repos.

Here, it was about eliminating global variables and making methods/functions closed.

Reply:
Huge improvement having dumped the global. +1